pkg/reconciler/ingress/ingress.go

```diff
@@ -461,3 +491,89 @@ func getLBStatus(gatewayServiceURL string) []v1alpha1.LoadBalancerIngressStatus
 func enableReconcileGateway(ctx context.Context) bool {
 	return config.FromContext(ctx).Network.AutoTLS || config.FromContext(ctx).Istio.ReconcileExternalGateway
 }
+
+func (r *Reconciler) reconcileRoutes(ctx context.Context, ia v1alpha1.Ingress, desiredRoutes []*routev1.Route) ([]*routev1.Route, error) {
+	routes, err := r.routeLister.List(labels.SelectorFromSet(map[string]string{
+		networking.IngressLabelKey:     string(ia.GetUID()),
+		serving.RouteLabelKey:          ia.GetLabels()[serving.RouteLabelKey],
+		serving.RouteNamespaceLabelKey: ia.GetLabels()[serving.RouteNamespaceLabelKey],
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list routes: %w", err)
+	}
+
+	// Create a map of all existing routes for quick access.
+	existingRoutes := make(map[string]*routev1.Route, len(routes))
+	for _, route := range routes {
+		existingRoutes[route.Namespace+"/"+route.Name] = route
```

Comment:
Can we please use `types.NamespacedName` instead of a concatenated `string`. I was lazy with the fix in the other repo, sorry for that.

Reply:
Sure, fixed. I was wondering it :smile: